postgresql/src/main/scala/org/apache/spark/sql/catalyst/expressionsSQLBuilder.scala

```diff
@@ -0,0 +1,244 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * Modifications and adaptations - Copyright (C) 2015 Stratio (http://stratio.com)
+ */
+
+package org.apache.spark.sql.catalyst
+
+import javax.xml.bind.DatatypeConverter
+
+import org.apache.spark.sql.catalyst.expressions.{Abs, Alias, And, Ascending, AttributeReference, BinaryExpression, BinaryOperator, CaseWhen, Cast, Concat, ConcatWs, Contains, Descending, EndsWith, Expression, GetStructField, If, In, InSet, IsNotNull, IsNull, Literal, MonotonicallyIncreasingID, NamedExpression, Not, Or, Pmod, SortDirection, StartsWith, StringRegexExpression, TernaryExpression, UnaryExpression, UnaryMathExpression, UnaryMinus, UnaryPositive}
+import org.apache.spark.sql.catalyst.expressions.aggregate.{AggregateExpression, AggregateFunction, Average}
+import org.apache.spark.sql.catalyst.plans._
+import org.apache.spark.sql.catalyst.plans.logical.{Aggregate, LogicalPlan}
+import org.apache.spark.sql.catalyst.util.DateTimeUtils
+import org.apache.spark.sql.types._
+import org.apache.spark.unsafe.types.UTF8String
+
+// TODO remove when upgrading to Spark2.0: this methods have already been implemented in Spark2.0
+object expressionsSQLBuilder {
+
+  case class AliasContext(aliasSupported: Boolean = true, logicalPlan: Option[LogicalPlan] = None)
+
+  def prettyName(any: Any): String = any.getClass.getSimpleName.toLowerCase
+
+  def partFunctionNamedExpressionToSQL(aliasContext: AliasContext): PartialFunction[Expression, String] = {
+    case attributeReference: AttributeReference =>
+
+      val name = if (aliasContext.aliasSupported || aliasContext.logicalPlan.isEmpty) {
+        attributeReference.name
+      } else {
+        aliasContext.logicalPlan.get.collectFirst {
+          // postgres does not support alias in the group by, so the alias in the aggregation is replaced by the sql of the expression
+          // TODO use pattern matching + unapply instead
+          case Aggregate(_, aggregateExpressions, _) =>
+            aggregateExpressions.collectFirst {
+              case Alias(aExp: Expression, aliasName) if attributeReference.name == aliasName =>
+                aExp.sql
+            } getOrElse attributeReference.name
+          // TODO case Project(...) => check whether the project contains alias or not
+        } getOrElse attributeReference.name
+      }
+
+      val qualifiersString =
+        if (attributeReference.qualifiers.isEmpty) "" else attributeReference.qualifiers.map("" + _ + "").mkString("", ".", ".")
+      s"$qualifiersString$name"
+
+    case al: Alias =>
+      val qualifiersString =
+        if (al.qualifiers.isEmpty) "" else al.qualifiers.map("" + _ + "").mkString("", ".", ".")
```

Comment:
I think the `if` expression isn't actually doing anything. Just keep the `else` branch body.

```
val qualifiersString = al.qualifiers.map("" + _ + "").mkString("", ".", ".")
```

Besides, the map doesn't seem to be doing anything.